src/m365/pa/commands/app/app-permission-ensure.ts

```diff
@@ -0,0 +1,204 @@
+
+interface Options extends GlobalOptions {
+  appName: string;
+  roleName: string;
+  userId?: string;
+  userName?: string;
+  groupId?: string;
+  groupName?: string;
+  tenant?: boolean;
+  asAdmin?: boolean;
+  environmentName?: string;
+  sendInvitationMail?: boolean;
+}
+
+class PaAppPermissionEnsureCommand extends PowerAppsCommand {
+  private static readonly roleNames = ['CanEdit', 'CanView'];
+  public get name(): string {
+    return commands.APP_PERMISSION_ENSURE;
+  }
+
+  public get description(): string {
+    return 'Assigns/updates permissions to a Power Apps app';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        groupId: typeof args.options.userId !== 'undefined',
+        groupName: typeof args.options.userName !== 'undefined',
+        tenant: !!args.options.tenant,
+        asAdmin: !!args.options.asAdmin,
+        environmentName: typeof args.options.environmentName !== 'undefined',
+        sendInvitationMail: !!args.options.sendInvitationMail
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--appName <appName>'
+      },
+      {
+        option: '--roleName <roleName>',
+        autocomplete: PaAppPermissionEnsureCommand.roleNames
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '--groupId [groupId]'
+      },
+      {
+        option: '--groupName [groupName]'
+      },
+      {
+        option: '--tenant'
+      },
+      {
+        option: '--asAdmin'
+      },
+      {
+        option: '-e, --environmentName [environmentName]'
+      },
+      {
+        option: '--sendInvitationMail'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.appName)) {
+          return `${args.options.appName} is not a valid GUID`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.roleName && PaAppPermissionEnsureCommand.roleNames.indexOf(args.options.roleName) < 0) {
+          return `${args.options.roleName} is not a valid roleName. Allowed values are ${PaAppPermissionEnsureCommand.roleNames.join(', ')}`;
+        }
+
+        if (args.options.environmentName && !args.options.asAdmin) {
```

Comment:
Let's also add a check that `environmentName` is filled in when using `asAdmin`.